### PR TITLE
Hyperzine prevents Diona regeneration. Hyperzine no longer speeds up Diona as much as it used to. 

### PR DESCRIPTION
**Pull request by Anton-Kr**
Tired of green fuckers running to your chem station every round begging for their pills to make them even more OP? Worry no longer.

# Changelog
:cl:
* tweak: Hyperizine prevents Diona health regeneration
* tweak: Hyperizine speedups for Diona are nowhere near as powerful as they were before.

### CODE REVIEW

**Comment by ultraplen** [comment]
A fair nerf. Possibly a bit harsh.

**Comment by BobdaBiscuit** [comment]
If only you knew what kind of hell you are inviting in by doing this.

**Comment by AutismSpider** [comment]
Better ways to do this to bad no one gives a fuck.

**Comment by Kurfursten** [comment]
With the current numbers, this will crit a Diona in 13 seconds, consuming 0.39u Hyperzine

The low lethal dosage should be taken into account when considering splash/smoke mechanics

**Comment by PJB3005** [comment]
Yeah the damage is WAY over the top.

Make it more manageable but still not something you want to consistently use.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -2365,7 +2365,13 @@
 
 	if(..())
 		return 1
-
+	
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.species.name == "Diona")
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Reagents.dm]: Use `isdiona(H)`

**Review by PJB3005 — DISMISSED** [review]

**Comment by Fruktlimpa** [comment]
Dionae is a pretty shit race lets not lie. The massive slowdown would make them extremely annoying to play which is why everyone chugs hyperzine in the first place.
Instead of making hyperzine a death sentence or forcing them to chug other pills at the same rate. Isn't it possible to make them slightly slower than a human by consuming hyperzine?
I realize it isn't ideal because it would still force them to chug hyperzine but at least then they wouldn't be speedy trees of no brute damage.

**Comment by Anton-Kr** [comment]
Im going to make the damage dealt a lot more reasonable (and potentially counteracted with chems) and also make the speed up to not be as great as it is for other species.

**Comment by Anton-Kr** [comment]
Will also try to make regeneration not work for diona while on hyperzine in which case Ill probably reduce damage dealt even more so

**Comment by BobdaBiscuit** [comment]
IMO, if you make hyperzine not make dionas sanic, and then disable their regen while under it's effects the damage is unneeded.

**Comment by Ephx1** [comment]
_It begins!_

**Comment by Blithering** [comment]
Disabling the regen when hyperzine is present as @BobdaBiscuit suggested seems like a fair tradeoff to me, not sure how feasible that is code wise though

**Comment by BobdaBiscuit** [comment]
It's perfectly possible, I had a look through the code for em and it's just a isdiona() + has hyperzine test and we're gucci

**Comment by Kurfursten** [comment]
I think disabling the regeneration is an interesting idea, because combat is when you would most want regeneration AND movement speed, and it's not trivial for most jobs to remove the hyperzine when combat is over unless you're carrying a pAI to smoke it out.

Overall, I appreciate its simplicity and balance as an idea.

**Comment by Anton-Kr** [comment]
Alright, so no damage dealt to diona, but no regen, and speed buff is nerfed.

**Comment by Exxion** [comment]
If it's less effective AND prevents regeneration, is the damage really necessary?

**Comment by Anton-Kr** [comment]
@PJB3005 Let me know if there are any other issues.

**Comment by Anton-Kr** [comment]
@Exxion It isn't. I forgot to remove it in last commit.

**Comment by gbasood** [comment]
>Dionae is a pretty shit race lets not lie. 

Yeah man regenerating through anything but burn damage
So shit

**Comment by Exxion** [comment]
And taking enough burn damage to be critted in two (2) standard laser shots

**Comment by aceedex** [comment]
maybe speed up diona a bit in exchange then, because I feel its unplayable without hyperzine

**Comment by Exxion** [comment]
Consider these options:
* Git Gud

**Comment by gbasood** [comment]
I'd rather make a balance change that's overkill and then find a halfway point afterwards than fiddle around with multiple PRs that are "too harsh man" according to a small subset of players

**Comment by ghost** [comment]
just make hyperzine heat up bodies enough to deal burn if you overdose

**Comment by Fruktlimpa** [comment]
@gbasood 
>Yeah man regenerating through anything but burn damage
So shit

I meant more in a sense that they are a stupid meme race that literally cannot function without hyperzine. Anyone who would willingly play Dionae without so much as a drop of hyperzine is insane.

**Comment by ghost** [comment]
I think this PR is bad because it changes the two main diona things, which is being slow and having high regen. Diona already suck, even with non nerfed hyperzine. Here is my ideas to fix diona, and make them more like tree people they should be.
>slightly faster base movement speed (still slower than human)
>unaffected by zas
>chemicals don't work on them

No zas, can't be chloral or sleepypen meme'd but they're stuck being sorta slow forever because they can't eat hyperzine.

**Comment by Shadowmech88** [comment]
I keep seeing people claim that dionae are OP.

Now, I haven't played in a while, so I'm honestly asking: are there a significant number of players rolling dionae? Something like 30%-40% at least?
Because that's what I would expect to happen if we had a race that was truly overpowered, that a significant number of people would be playing it for the advantage it offered.

**Comment by BobdaBiscuit** [comment]
Actually we had a week were the majority of sec teams were all diona, is was a pretty funny meme.

**Comment by SonixApache** [comment]
if you're going to take away their regen you might as well make them as fast as humans
level the playing field rather than home advantage

and while we're at it, antitox ODing when?

**Comment by Anton-Kr** [comment]
Regen is still there but only when they have no hyperzine in system

**Comment by SonixApache** [comment]
Yes, that's the point. If they don't have their bullshit regeneration anymore, why make them slower still than humans on drugs?

**Comment by FudgePucker** [comment]
Number one murder bone weapon is egun.
Zap and laser to death.
Number one lethal option security has is egun.
Zap and laser to death.
Everybody has a toolbelt with a welder.
Stuns are king burn damage is second.

Hell they can't survive polyacid smoke grenades. How are they super OP again?

I like Solarious' idea. They are so slow it takes half a round to cross the station, you take hyperzine or roleplay a tree. A minor speed nerf instead of a super major "oh god how can it be possible to move this slow" one. Nerf regen so you can kill one in full light with a esword and then we have balance.

**Comment by Blithering** [comment]
Combat isn't the only factor when considering a race's strengths. Remember they don't need to breathe or eat, are unaffected by pressure, heal from radiation, can shrug off ZAS and ignore overdose limits on meds. Surviving as a diona is a lot simpler because you can pretty much ignore everything that isn't fire.

**Comment by Exxion** [comment]
At the cost of walking at roughly the same speed as a slime mold.
Which is why hyperzine is too good for them, which is why this PR exists

Also, all of you fucks saying that dionae are unplayably slow need to either get more patient or stop playing dionae

**Comment by Intigracy** [comment]
Clearly we should just make dionae unable to move.

**Comment by Shadowmech88** [comment]
They are trees, after all. Trees don't move.

**Comment by Intigracy** [comment]
wait sorry wrong meme

the obvious solution is to remove skeletons

**Comment by DamianX** [comment]
Fix the changelog already

**Review by PJB3005 — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 2]
```diff
@@ -76,6 +76,8 @@
 	if(reagents.has_reagent(HYPERZINE))
 		if(dna.mutantrace == "slime")
 			tally *= 2
+		if(isdiona(src))
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/human_movement.dm]: Use `else if`.

File: code/modules/mob/living/carbon/human/life/handle_chemicals_in_body.dm [hunk 3]
```diff
@@ -42,7 +42,7 @@
 		if(species.flags & IS_PLANT)
 			if(nutrition > 500)
 				nutrition = 500
-			if(light_amount >= 3) //If there's enough light, heal
+			if(light_amount >= 3 & !reagents.has_reagent(HYPERZINE)) //If there's enough light, and you do not have hyperzine in body, heal
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/human/life/handle_chemicals_in_body.dm]: Use `&&`, `&` is a bitwise operation.

**Review by ComicIronic — CHANGES_REQUESTED** [review]
Needs fixing.

**Comment by Anton-Kr** [comment]
@ComicIronic done

**Comment by BobdaBiscuit** [comment]
How fast is it ingame compared to the human base speed? Cause the tally boost seems hella lower.

**Comment by Anton-Kr** [comment]
@BobdaBiscuit Human slowdown by default is 0. Diona is 7 without hyperzine, and 3 with, so somewhat slower then humans still.

**Comment by BobdaBiscuit** [comment]
Eh if it's still too slow I suppose another pr can be made to tweak it.

**Review by ComicIronic — APPROVED** [review]

**Comment by Anton-Kr** [comment]
MERGE WHEN

**Comment by BobdaBiscuit** [comment]
\>"fair" diona nerf that most people agree with
\>still not merged nearly a month later

REALLY MAKES YOU THINK

**Comment by Intigracy** [comment]
If this needs adjusting we can do so.

**Pull request merged**